### PR TITLE
Allow {FileSystem,Http}Repository to customize metadata/targets prefix

### LINKED ISSUE #182: Customize {FileSystem,Http}Repository directory structure
FileSystemRepository assumes that metadata is stored under `metadata/` and targets are stored under `targets/`. HttpRepository allows for metadata to be customized, but assumes targets are found at the top level. We should allow this to be customized since some projects may structure their repositories differently (for example, go-tuf puts their metadata in the top level directory).

### PR DESCRIPTION
**Pull request by erickt**
This allows users to target repositories with alternative layouts.

Closes #182

### CODE REVIEW

File: src/repository.rs [hunk 1]
```diff
@@ -92,12 +92,74 @@ where
     }
 }
 
+
+/// A builde rto create a repository contained on the local file system.
```
Comment (heartsucker) [inline src/repository.rs]: Typo: `builde rto` -> `builder to`

**Review by heartsucker — COMMENTED** [review]
Minor need. Needs a rebase too.

File: src/repository.rs (hunk 1 above)
Comment (erickt) [reply src/repository.rs]: fixed, thanks

**Comment by erickt** [comment]
Rebased (including being on top of #192). Thanks for all the reviews!

**Review by heartsucker — APPROVED** [review]

**Pull request merged**